lib/pavilion/schedulers/plugins/flux.py

```diff
@@ -0,0 +1,332 @@
+            "exception.note": ("exception_note",),
+            "runtime": ("t_run", "t_cleanup"),
+            "status": ("state", "result"),
+            "status_abbrev": ("state", "result"),
+            "t_remaining": ("expiration", "state", "result"),
+        }
+
+        # Set job attributes we will use later to get job statuses
+        self.job_attrs = set()
+        for field in self.jobs_formatter.fields:
+            if field not in fields2attrs:
+                self.job_attrs.update((field,))
+            else:
+                self.job_attrs.update(fields2attrs[field])
+
+
+    def _kickoff(self, pav_cfg, job: Job, sched_config: dict) -> JobInfo:
+        """Submit the kick off script using sbatch."""
+
+        self._prepare_job_formatter()
+        output = job.sched_log.as_posix()
+        error = job.sched_log.as_posix() + ".err"
+
+        # Generate the flux job
+        # Assume the filename includes a hashbang
+        # flux does not support mem_mb, disk_mb
+        fluxjob = JobspecV1.from_command(
+            command=["/bin/bash", job.kickoff_path.as_posix()],
+            num_tasks=sched_config['tasks_per_node'] or 1,
+        )
+        
+        # A duration of zero (the default) means unlimited
+        fluxjob.duration = sched_config['time_limit'] or 0
+        fluxjob.stdout = output
+        fluxjob.stderr = error
+
+        # This doesn't seem to be used?
+        fluxjob.cwd = str(job.tests_path)
+        fluxjob.environment = dict(os.environ)
+        flux_future = self._fexecutor.submit(fluxjob)
+        
+        job._jobid = str(flux_future.jobid())
+        job._submit_time = time.time()
+        job._flux_future = flux_future
+        sys_name = sys_vars.get_vars(True)['sys_name']
+
+        return JobInfo({
+            'id': str(flux_future.jobid()),
+            'sys_name': sys_name,
+        })        
+
+
+    def _scontrol_show(self, *args, timeout=10):
+        """Run scontrol show and return the parsed output.
+
+        :param list(str) args: Additional args to scontrol.
+        :param int timeout: How long to wait for results.
+        """
+        print('SCONTROL SHOW')
+        import IPython
+        IPython.embed()
+
+        cmd = ['scontrol', 'show'] + list(args)
+
+        proc = subprocess.Popen(cmd,
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+
+        try:
+            stdout, stderr = proc.communicate(timeout=timeout)
+        except subprocess.TimeoutExpired:
+            return []
+
+        stdout = stdout.decode('utf8')
+        stderr = stderr.decode('utf8')
+
+        if proc.poll() != 0:
+            raise ValueError(stderr)
+
+        results = []
+        for section in stdout.split('\n\n'):
+            try:
+                results.append(self._scontrol_parse(section))
+            except (KeyError, ValueError):
+                pass
+
+        return results
```

Comment:
Not necessary for Flux.

Reply:
haha yes I know, this is left over from before.